keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md

```diff
@@ -235,51 +452,101 @@ _This section must be completed when targeting beta graduation to a release._
 
 * **How can a rollout fail? Can it impact already running workloads?**
 
-  TBD for beta.
+The primary method of rollout failure is due to configuration issues. When
+migrating to use external CCMs there are several changes which must be made
+to ensure proper functioning. When encountering a failure, users should review
+the migration documentation to ensure that all command line flags, RBAC
+requirements, taint tolerations, and manifests are following the best
+practices.
+
+It is possible that a failed rollout could affect running workloads. Normal
+functioning of external CCMs involves activities around labeling new node
+objects, removing node objects, and configuring loadbalancer type services.
+If a rollout of external CCMs failed, users might see failures of workloads
+to schedule, workloads might be evicted, and service-based traffic could be
+interrupted.
 
 * **What specific metrics should inform a rollback?**
 
-   TBD for beta.
+There are a few observable behaviors that inform a rollback. If users start to
+see nodes failing to register, load balancer calls failing or not happening,
+deleted nodes not being removed, or components failing to start due to missing
+taint tolerations, these could all be possible signs that a CCM rollout has
+failed.
+
+In addition there are several metrics exposed by the cloud-provider framework
+that is maintained by SIG Cloud Provider. If the external CCM is using the
+cloud-provider framework, the following metrics could inform about failing
+behavior, users should be observant of these metric values increasing over
+a short period of time, or after upgrade:
+
+* `cloud_provider_taint_removal_delay_seconds`
+* `initial_node_sync_delay_seconds`
+* `nodesync_error_total`
+* `nodesync_latency_seconds`
+* `update_loadbalancer_host_latency_seconds`
 
 * **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
 
-  TBD for beta.
+  * Yes, tested manually by various providers, notably AWS and GCE.
```

Comment:
Can you please be more specific? Any links?

Reply:
i don't think we have links, but i will ask.

Reply:
added a little more here, let me know if that works.